File: ts/project/.sdk/tm/rb/utility/make_fetch_def.rb
# ProjectName SDK utility: make_fetch_def
require_relative 'struct/voxgig_struct'
require_relative '../core/result'
module ProjectNameUtilities
  MakeFetchDef = ->(ctx) {
    spec = ctx.spec
    return nil, ctx.make_error("fetchdef_no_spec", "Expected context spec property to be defined.") unless spec

    ctx.result = ProjectNameResult.new({}) unless ctx.result
    spec.step = "prepare"

    url, err = ctx.utility.make_url.call(ctx)
    return nil, err if err

    spec.url = url

    fetchdef = { "url" => url, "method" => spec.method, "headers" => spec.headers }
    if spec.body
      fetchdef["body"] = spec.body.is_a?(Hash) ? VoxgigStruct.jsonify(spec.body) : spec.body
    end

    return fetchdef, nil
  }
end

File: ts/project/.sdk/tm/py/utility/make_fetch_def.py
# ProjectName SDK utility: make_fetch_def

from __future__ import annotations
from utility.voxgig_struct import voxgig_struct as vs


def make_fetch_def_util(ctx):
    spec = ctx.spec
    if spec is None:
        return None, ctx.make_error("fetchdef_no_spec",
            "Expected context spec property to be defined.")

    from core.result import ProjectNameResult
    if ctx.result is None:
        ctx.result = ProjectNameResult({})

    spec.step = "prepare"

    url, err = ctx.utility.make_url(ctx)
    if err is not None:
        return None, err

    spec.url = url

    fetchdef = {
        "url": url,
        "method": spec.method,
        "headers": spec.headers,
    }

    if spec.body is not None:
        if isinstance(spec.body, dict):
            fetchdef["body"] = vs.jsonify(spec.body)
        else:
            fetchdef["body"] = spec.body

    return fetchdef, None
